packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -74,10 +77,18 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 		[]
 	);
 
-	const blockPatterns = useMemo(
-		() => unionBy( settingsBlockPatterns, restBlockPatterns, 'name' ),
-		[ settingsBlockPatterns, restBlockPatterns ]
-	);
+	const blockPatterns = useMemo( () => {
```

Comment:
I'm not sure this is the right place to do this(editor settings). I haven't tested this but an example could be in a `page` insert a `Query Loop` that shows let's say `book` post type. Shouldn't we show the related patterns there?